clients/js/genjs.go

```diff
@@ -0,0 +1,249 @@
+package jsclient
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Clever/wag/swagger"
+	"github.com/Clever/wag/templates"
+	"github.com/go-openapi/spec"
+)
+
+// Generate generates a client
+func Generate(modulePath string, s spec.Swagger) error {
+	pkgName, _ := s.Info.Extensions.GetString("x-npm-package")
+
+	tmplInfo := clientCodeTemplate{
+		ClassName:   camelCase(s.Info.InfoProps.Title, true),
+		PackageName: pkgName,
+		ServiceName: s.Info.InfoProps.Title,
+		Version:     s.Info.InfoProps.Version,
+		Description: s.Info.InfoProps.Description,
+	}
+
+	for _, path := range swagger.SortedPathItemKeys(s.Paths.Paths) {
+		pathItem := s.Paths.Paths[path]
+		pathItemOps := swagger.PathItemOperations(pathItem)
+		for _, method := range swagger.SortedOperationsKeys(pathItemOps) {
+			op := pathItemOps[method]
+			if op.Deprecated {
+				continue
+			}
+			methodCode, err := methodCode(op, method, s.BasePath, path)
+			if err != nil {
+				return err
+			}
+			tmplInfo.Methods = append(tmplInfo.Methods, methodCode)
+		}
+	}
+
+	indexJS, err := templates.WriteTemplate(indexJSTmplStr, tmplInfo)
+	if err != nil {
+		return err
+	}
+
+	packageJSON, err := templates.WriteTemplate(packageJSONTmplStr, tmplInfo)
+	if err != nil {
+		return err
+	}
+
+	if err = ioutil.WriteFile(filepath.Join(modulePath, "index.js"), []byte(indexJS), 0644); err != nil {
+		return err
+	}
+
+	if err = ioutil.WriteFile(filepath.Join(modulePath, "package.json"), []byte(packageJSON), 0644); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type clientCodeTemplate struct {
+	PackageName string /* Name of the npm package */
+	ClassName   string /* Name of the class to export */
+	ServiceName string /* Service's internal name (aka the github repo/discovery name) */
```

Comment:
Decided to channel my inner Abhinav and just remove the comments since I don't think they added much.